src/main/java/bio/overture/ego/security/JWTAuthorizationFilter.java

```diff
@@ -69,7 +69,7 @@ public void doFilterInternal(
     }
     val tokenPayload = request.getHeader(HttpHeaders.AUTHORIZATION);
 
-    if (tokenPayload != null && tokenPayload.startsWith(applicationService.APP_TOKEN_PREFIX)) {
+    if (tokenPayload != null && tokenPayload.startsWith(BasicAuthToken.TOKEN_PREFIX)) {
```

Comment:
This is incorrect as this turns Basic tokens in Bearer tokens. APP_TOKEN_PREFIX is declared as the following in the ApplicationService class:
```java
public static final String APP_TOKEN_PREFIX = "Basic ";
```

This would break the client credentials OAuth2 flow as Basic auth would be rejected.

I just verified and the tests pass which is scary given this particular line, so lets fix this and add a test for that. 
The test should do the following:
 1. Have an app registered to ego
 2. Use client credentials with basic token to get application JWT.
 3. Verify that is was successful

Reply:
ignore me